### DISCUSSION TITLE
Transform ordering

### DISCUSSION THREAD

**Discussion by cark** [bevyengine/bevy#4486] [Q&A] [answered]
In my 2D game, I'm trying to first scale a sprite, then rotate it, but it looks like the Transform will first apply its rotation then do the scaling.

The use case is this: 
I'm drawing a sword hit. Depending on character progression, the length (range) of that hit will vary.
So I'm scaling the sprite in one dimension only (The x direction in this case).
Then I need to apply a rotation to that hit so that it faces the enemy.

Using Transform::with_scale and Transform_with_rotation in any order will result in the same squished sprite when rotation is non-zero.

Is there a way to specify the order of those transformations ?

I want to note that I had some better success using glam matrices and Transform::from_matrix, but then my child heron collider doesn't seem to understand what's going on.

**Comment by rparrett** [comment, marked as answer]
There is a bug in Bevy 0.6 that caused sprites specifically to do rotation first.

This was fixed by [this PR](https://github.com/bevyengine/bevy/pull/3811), which will be included in Bevy 0.7 ~~which is very close to release~~ [was just released](https://bevyengine.org/news/bevy-0-7/).

**Reply by cark** [reply]
Ah I was pretty sure I couldn't be the only one with that issue. Thanks !
